Directory.Build.props

```diff
@@ -10,4 +10,45 @@
   <PropertyGroup Condition="'$(StabilizePackageVersion)' == 'true'">
     <StableVersion>$(VersionPrefix)</StableVersion>
   </PropertyGroup>
+
+  <PropertyGroup>
+    <_hostArch>$([System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture.ToString().ToLowerInvariant)</_hostArch>
+    <BuildArchitecture>$(_hostArch)</BuildArchitecture>
+  </PropertyGroup>
+
+    <!-- Common properties -->
+  <PropertyGroup>
+
+    <__ProjectDir Condition="'$(__ProjectDir)'==''">$(MSBuildThisFileDirectory)</__ProjectDir>
+    <ProjectDir>$(__ProjectDir)\</ProjectDir>
+    <RootRepoDir>$(ProjectDir)</RootRepoDir>
+    <ProjectDir Condition="'$(__ProjectDir)'==''">$(MSBuildThisFileDirectory)</ProjectDir>
+
+    <BaseIntermediateOutputPath>$(RootRepoDir)artifacts\obj\$(PlatformConfigPathPart)\</BaseIntermediateOutputPath>
```

Comment:
Just for future reference really: `$([MSBuild]::NormalizeDirectory(..))` - this will use the correct slash, and ensure the trailing slash.